src/Framework/NativeMethods.cs

```diff
@@ -709,10 +711,10 @@ internal static bool IsMono
 #if !CLR2COMPATIBILITY
     private static bool? _isWindows;
 #endif
-
     /// <summary>
     /// Gets a flag indicating if we are running under some version of Windows
     /// </summary>
+    [SupportedOSPlatformGuard("windows")]
```

Comment:
I don't think you can do this? Doesn't this mean that platforms other than windows can't check if they're windows or not?